### PR TITLE
Added Console App support for RTM DotNetCore

### PR DESCRIPTION
**Pull request by saurabsa**
- New prompt for users to select console or web
    projects for DotNetCore project
 - No more port related data in docker assets for console app
 - Added tests for RTM console .Net Core project

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ var isGoWeb = false;
 var baseImageName = '';
 var updateProjectJsonNoteForUser = null;
 var updateProgramCSNoteForUser = null;
+var isDotNetWeb = true;
```
Comment (NCarlsonMSFT) [inline generators/app/index.js]: Could we unify this with isGoWeb?

Comment (NCarlsonMSFT) [inline generators/app/index.js]: Is this being logged in the telemetry?

---
In reply to: [71039209](https://github.com/Microsoft/generator-docker/pull/103#discussion_r71039209) [](ancestors = 71039209)

**Comment by NCarlsonMSFT** [comment]
:clock1:

File: generators/app/index.js [hunk 2]
```diff
@@ -102,7 +100,7 @@ function showPrompts() {
             },
             when: function (answers) {
                 // Show this answer if user picked .NET, Node.js or Golang that's using a web server.
-                return answers.projectType === 'dotnet' || answers.projectType === 'nodejs' || (answers.projectType === 'golang' && answers.isGoWeb);
+                return (answers.projectType === 'dotnet' || answers.projectType === 'nodejs' || answers.projectType === 'golang') && answers.isWebProject;
```
Comment (NCarlsonMSFT) [inline generators/app/index.js]: >(answers.projectType === 'dotnet' || answers.projectType === 'nodejs' || answers.projectType === 'golang') && [](start = 23, length = 109)

You can just cut all of the project type checking

File: generators/app/index.js [hunk 3]
```diff
@@ -168,6 +166,7 @@ function handleNodeJs(yo) {
     templateData.volumeMap = '.:/src';
     templateData.includeComposeForDebug = true;
     templateData.debugPortNumber = "5858";
+    templateData.isWebProject = isWebProject;
```
Comment (NCarlsonMSFT) [inline generators/app/index.js]: Can't this be handled in getDefaultTemplateData?

**Comment by NCarlsonMSFT** [comment]
I don't see any updates to the compose file, does the node debugging port still get outputted in the non-web case?

File: test/nodejstests.js [hunk 4]
```diff
@@ -9,12 +9,100 @@ var path = require('path');
...
+    it('correct dockerfile contents (release)', function (done) {
+        assert.noFileContent('Dockerfile', 'nodemon');
+        assert.fileContent('Dockerfile', 'ENTRYPOINT ["npm", "start"]');
+        done();
+    });
+
+    it('correct compose file contents (debug)', function (done) {
+        assert.fileContent('docker-compose.debug.yml', 'image: testimagename:debug');
+        assert.fileContent('docker-compose.debug.yml', '.:/src');
+        assert.noFileContent('docker-compose.debug.yml', '"3000:3000"');
```
Comment (NCarlsonMSFT) [inline test/nodejstests.js]: >assert.noFileContent('docker-compose.debug.yml', '"3000:3000"'); [](start = 8, length = 64)

We should assert that this does have 5858:5858 and "ports:"

File: test/nodejstests.js [hunk 5]
```diff
@@ -9,12 +9,100 @@ var path = require('path');
...
+        assert.fileContent('docker-compose.debug.yml', '- REMOTE_DEBUGGING');
+        done();
+    });
+
+    it('correct compose file contents (release)', function (done) {
+        assert.fileContent('docker-compose.yml', 'image: testimagename');
+        assert.noFileContent('docker-compose.yml', '.:/src');
+        assert.noFileContent('docker-compose.yml', '"3000:3000"');
```
Comment (NCarlsonMSFT) [inline test/nodejstests.js]: >assert.noFileContent('docker-compose.yml', '"3000:3000"'); [](start = 8, length = 58)

We should assert that this does not have 5858:5858 nor "ports:"

File: test/nodejstests.js (hunk 4 above)
Comment (NCarlsonMSFT) [inline test/nodejstests.js]: Could you also add this to the web case, it is a test we missed previously.

---
In reply to: [71217001](https://github.com/Microsoft/generator-docker/pull/103#discussion_r71217001) [](ancestors = 71217001)

File: test/nodejstests.js [hunk 6]
```diff
@@ -9,12 +9,100 @@ var path = require('path');
...
+
+    it('Correct script file contents (bash)', function (done) {
+        assert.fileContent('dockerTask.sh', 'isWebProject=false');
+        assert.noFileContent('dockerTask.sh', 'dotnet publish');
+        assert.fileContent('dockerTask.sh', 'composeForDebug');
+        assert.noFileContent('dockerTask.sh', 'startDebugging');
+        done();
+    });
+
+    it('correct dockerfile contents (debug)', function (done) {
```
Comment (NCarlsonMSFT) [inline test/nodejstests.js]: >it('correct dockerfile contents (debug)', function (done) { [](start = 4, length = 59)

For all of the nodes tests: we should also be checking the exposes for the dockerfiles, not just the port mappings in the compose files.

File: generators/app/templates/docker-compose.yml [hunk 7]
```diff
@@ -5,8 +5,8 @@ services:
     image: <%= imageName %><% if (environment === 'debug') { %>:debug<% } %>
     build:
       context: .
-      dockerfile: Dockerfile<% if (environment === 'debug') { %>.debug<% } %><% if (isWebProject && portNumber) { %>
-    ports:
+      dockerfile: Dockerfile<% if (environment === 'debug') { %>.debug<% } %><% if ((isWebProject && portNumber) || (environment === 'debug' && debugPortNumber)) { %>
```
Comment (NCarlsonMSFT) [inline generators/app/templates/docker-compose.yml]: >(environment === 'debug' && debugPortNumber) [](start = 116, length = 44)

Nitpick: this is a different way of doing the if than below. If both work please pick one and make them consistent (I favor this one as less verbose)

Comment (saurabsa) [inline generators/app/templates/docker-compose.yml]: The below (environment === 'debug') has a broader block containing check for volumeMap and includeComposeForDebug along with debugPortNumber.

---
In reply to: [71240674](https://github.com/Microsoft/generator-docker/pull/103#discussion_r71240674) [](ancestors = 71240674)

Comment (NCarlsonMSFT) [inline generators/app/templates/docker-compose.yml]: Good point

---
In reply to: [71241206](https://github.com/Microsoft/generator-docker/pull/103#discussion_r71241206) [](ancestors = 71241206,71240674)

File: generators/app/templates/_dockerTaskGeneric.ps1 [hunk 8]
```diff
@@ -41,9 +41,9 @@ $imageName="<%= imageName %>"
 $projectName="<%= composeProjectName %>"<% if (includeStartDebugging) { %>
 $serviceName="<%= serviceName %>"
 $containerName="<%= '${projectName}_${serviceName}' %>_1"<% } %>
+$isWebProject=$<%= isWebProject %><% if (isWebProject) { %>
 $publicPort=<%= portNumber %>
-$isWebProject=$<%= isWebProject %>
-$url="http://docker:$publicPort"<% if (projectType === 'dotnet' && (dotnetVersion === 'RC2' || dotnetVersion === 'RTM')) { %>
```
Comment (NCarlsonMSFT) [inline generators/app/templates/_dockerTaskGeneric.ps1]: >$url [](start = 0, length = 4)

This is still being referenced below. If you're going to if out web stuff for the console case you need to get it all.

File: generators/app/templates/_dockerTaskGeneric.ps1 [hunk 9]
```diff
@@ -106,16 +105,16 @@ function Compose () {
     }
 }<% if (includeStartDebugging) { %>
 
-function StartDebugging () {
+function StartDebugging () {<% if (isWebProject) { %>
     Write-Host "Running on $url"
-
+    <% } %>
```
Comment (NCarlsonMSFT) [inline generators/app/templates/_dockerTaskGeneric.ps1]: >     [](start = 0, length = 4)

Stray spaces

**Comment by NCarlsonMSFT** [comment]
>         assert.fileContent('dockerTask.ps1', '$isWebProject=$true');

These checks should be replaced with check for whether or not OpenSite  is in the script files

---
Refers to: test/dotnettests.js:145 in 1ed783a. [](commit_id = 1ed783a94c40e829f1d2db1b9b17357ecbf2e277, deletion_comment = True)

**Comment by saurabsa** [comment]
>         assert.fileContent('dockerTask.ps1', '$isWebProject=$true');

Yes. makes sense. Thanks

---
In reply to: [233691293](https://github.com/Microsoft/generator-docker/pull/103#issuecomment-233691293) [](ancestors = 233691293)

---
Refers to: test/dotnettests.js:145 in 1ed783a. [](commit_id = 1ed783a94c40e829f1d2db1b9b17357ecbf2e277, deletion_comment = True)

**Pull request merged**